Issue/6

yhmane opened the pull request.

Commit:
refactor: Mockito -> BDDMockito로 변경

Commit:
refactor: Mockito -> BDDMockito로 변경

Commit:
refactor: Store method name 변경

Commit:
refactor: Store method name 변경

0hun committed:
feat (feature) : 회원 수정 메소드 추가

회원 수정 메소드 추가
  - 회원 수정 메소드 추가
  - BDDMockito 적용하여 테스트 코드 수정

0hun commented on src/main/java/com/delivery/store/controller/StoreController.java:
```diff
@@ -23,24 +23,25 @@
     private final StoreService storeService;
 
     @GetMapping("/{storeId}")
-    public StoreResponseDto findStore(@PathVariable Long storeId) {
```
primitive type 타입으로 변경해주면 좋을 거 같습니다.

0hun commented on src/main/java/com/delivery/store/controller/StoreController.java:
```diff
@@ -23,24 +23,25 @@
-    public ResponseEntity<?> createStore(@RequestBody StoreRequestDto storeRequest) throws URISyntaxException {
-        StoreEntity store = storeService.createStore(storeRequest);
-        return ResponseEntity.created(new URI("/stores/" + store.getId())).body("{}");
+    public ResponseEntity<?> create(@RequestBody StoreRequestDto storeRequest) throws URISyntaxException {
```
StoreRequestDto에 참조 타입이 있다면 String 제외 나머지는 기본 타입으로 변경해주시면 좋을 것 같습니다.

0hun commented on src/main/java/com/delivery/store/controller/StoreController.java:
```diff
@@ -23,24 +23,25 @@
-        storeService.deleteStore(storeId);
+    public ResponseEntity<?> delete(@PathVariable Long storeId) {
+        storeService.delete(storeId);
         return ResponseEntity.status(HttpStatus.OK).build();
```
200 코드를 내보낼 경우에 status를 직접 지정하는 것 대신에 ResponseEntity.ok().build()와 같은 static 함수로 사용 가능합니다

0hun commented on src/main/java/com/delivery/store/controller/StoreController.java:
```diff
@@ -23,24 +23,25 @@
+        return ResponseEntity.created(new URI("/stores/" + store.getId())).build();
     }
 
     @PutMapping("/{storeId}")
```
url에 store/ <- 이렇게 store의 id라는 것을 알수 있으므로,  store/{id} 이렇게 지정해도 될 것 같습니다.

0hun replied:
https://docs.microsoft.com/ko-kr/azure/architecture/best-practices/api-design

rest api 설계에 관한 내용입니다. 참고할 만한 좋은 내용이니 공유합니다.

0hun committed:
refactor : Mockito을 BDDMockito으로 변경

Mockito을 BDDMockito으로 변경
- void method BDDMockito으로 변경
- exception 테스트 시 BDDMockito으로 변경

yhmane committed:
Merge pull request #16 from 0hun/issue/9

#9 회원 수정 메소드 추가

yhmane committed:
Merge branch 'issue/6' of https://github.com/yhmane/delivery-service into issue/6

yhmane committed:
issue/6 Controller 정리

0hun commented:
상위 url에 store가 있으니 "/{storeId}" 대신에 -> /{id}로 바꾸는 것이 의미 중복이 되지 않고 좋을 것 같습니다. 해당 내용은 다음 이슈 때 변경해주시면 좋을 것 같습니다. 머지 하겠습니다.
